### PR TITLE
Vacsina facade implementovana, zjedotene exceptions v service, hotove testy na category facade a user service

### PR DESCRIPTION
**Pull request by jkbpolacek**
Todo: testy na ItemFacade (god help me) a dokoncit obrazky do itemov + rozsirit item service testy ?

### CODE REVIEW

File: lost-and-found-service-service/src/main/java/cz/muni/fi/service/ItemServiceImpl.java [hunk 1]
```diff
@@ -24,55 +25,63 @@
...
+
+    @Autowired
+    LocationService locationService;
+
     @Override
-    public Item archiveItem(Item item) throws ServiceException{
+    public void archiveItem(Item item) throws ServiceException {
         if (item == null) {
-            throw new IllegalArgumentException("Item cannot be null");
+            throw new ServiceException("Item cannot be null");
```
Comment (jkbpolacek) [inline lost-and-found-service-service/src/main/java/cz/muni/fi/service/ItemServiceImpl.java]: exceptions v tychto "advanced metodach" povazujem ako jedine za diskutabilne.

Inak by podla mna service mal proste volat dao, a neriesit veci ktore uz riesi dao

File: lost-and-found-service-service/src/main/java/cz/muni/fi/service/ItemService.java [hunk 2]
```diff
@@ -20,90 +20,128 @@
...
+     */
+    void removeCategoryFromItem(Long itemId, Long categoryId) throws ServiceException;
+
+
+    /**
+     * Changes found location to location or null if locationId null
+     *
+     * @param itemId     of item to change location
+     * @param locationId null or location id
+     * @throws ServiceException when location with given id when id not null doesn't exist. or null itemId
```
Comment (TerkaSlaninakova) [inline lost-and-found-service-service/src/main/java/cz/muni/fi/service/ItemService.java]: this @throws and a few below don't make sense, please fix

File: lost-and-found-service-service/src/main/java/cz/muni/fi/service/ItemServiceImpl.java [hunk 3]
```diff
@@ -175,21 +173,112 @@ public Item getItembyId(Long id) {
...
+
+            item.setLostLocation(location);
+        } else {
+            item.setLostLocation(null);
+        }
+        updateItem(item);
+    }
+
+    @Override
+    public void changeUser(Long itemId, Long userId) {
```
Comment (TerkaSlaninakova) [inline lost-and-found-service-service/src/main/java/cz/muni/fi/service/ItemServiceImpl.java]: maybe rename to changeOwner?

File: lost-and-found-service-service/src/main/java/cz/muni/fi/service/facade/ItemFacadeImpl.java [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+        itemService.resolveFoundItem(
+                beanMappingService.mapTo(itemDTO, Item.class),
+                lostDate,
+                beanMappingService.mapTo(lostDate, Location.class),
+                beanMappingService.mapTo(owner, User.class));
+    }
+
+    @Override
+    public void changeImage(ItemChangeImageDTO itemChangeDTO) {
+        // TODO
```
Comment (TerkaSlaninakova) [inline lost-and-found-service-service/src/main/java/cz/muni/fi/service/facade/ItemFacadeImpl.java]: will this be done in M2?

File: lost-and-found-service-service/src/main/java/cz/muni/fi/service/ItemServiceImpl.java (hunk 1 above)
Comment (TerkaSlaninakova) [reply lost-and-found-service-service/src/main/java/cz/muni/fi/service/ItemServiceImpl.java]: suhlas, ze checky na veci co checkuje dao netreba, ale pri archivovani potrebujem checknut v akom je stave item a ak je nahodou null, nechcem aby mi to vyfailovalo na null dereferencii, tak si myslim, ze tam v tomto pripade check na null dava zmysel.

Comment (jkbpolacek) [reply lost-and-found-service-service/src/main/java/cz/muni/fi/service/ItemServiceImpl.java]: tie checky som tam nechal len som zmenil aky exception hadzu

File: lost-and-found-service-service/src/main/java/cz/muni/fi/service/facade/ItemFacadeImpl.java [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    @Override
+    public void addItemFound(ItemCreateDTO itemCreateDTO) {
+        ItemDTO itemDTO = beanMappingService.mapTo(itemCreateDTO, ItemDTO.class);
+        itemDTO.setStatus(Status.CLAIM_RECEIVED_FOUND);
+
+        addItem(itemDTO);
+    }
+
+    private void addItem(ItemDTO itemDTO) {
+        itemDTO.setFoundDate(LocalDate.now());
```
[the remark this answers is not in the document]
Comment (jkbpolacek) [reply lost-and-found-service-service/src/main/java/cz/muni/fi/service/facade/ItemFacadeImpl.java]: ok

**Pull request merged**